### PR TITLE
QuestionsView - ask & display list of questions

### PR DESCRIPTION
**Pull request by xgarrett**

### CODE REVIEW

File: src/actions/QuestionsActions.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        const question = { title, body };
+        dispatch(askQuestionRequest());
+
+        request('/api/questions', {
+            method: 'POST',
+            body: {
+                question,
+            },
+        })
+            .then(askQuestionSuccess())
```
Comment (FredZgred) [inline src/actions/QuestionsActions.js]: ?? is it work? It should be `.then(askQuestionSuccess)` or `.then(() => askQuestionSuccess())`

File: src/components/AskQuestion/index.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+function submit(e, props) {
+    e.preventDefault();
+
+    const { title, body } = e.target.elements;
+
+    props.onSubmit(title.value, body.value);
+}
+
+export default function AskQuestion(props) {
```
Comment (FredZgred) [inline src/components/AskQuestion/index.js]: Why didn't you make it class having submit method?

File: src/components/Question/icon_question.svg [hunk 3]
```diff
@@ -0,0 +1,4 @@
+<svg fill="#000000" height="36" viewBox="0 0 24 24" width="36" xmlns="http://www.w3.org/2000/svg">
+    <path d="M0 0h24v24H0z" fill="none"/>
+    <path d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zm2 16H8v-2h8v2zm0-4H8v-2h8v2zm-3-5V3.5L18.5 9H13z"/>
+</svg>
```
Comment (FredZgred) [inline src/components/Question/icon_question.svg]: enter :)

File: src/components/Question/index.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+const propTypes = {
+    title: PropTypes.string.isRequired,
+    ownerId: PropTypes.string.isRequired,
+    createDate: PropTypes.string.isRequired,
+    updateDate: PropTypes.string.isRequired,
+    onClick: PropTypes.func.isRequired,
+};
+
+export default function Question(props) {
```
Comment (FredZgred) [inline src/components/Question/index.js]: Isn't our convention like using destructuring?

File: src/utils/requester.js [hunk 5]
```diff
@@ -17,7 +17,7 @@ export default function request(url, config) {
     return fetch(url, config)
         .then(response => {
             if (response.status >= 200 && response.status < 300) {
-                return response.json();
+                return Promise.resolve(response.json());
```
Comment (FredZgred) [inline src/utils/requester.js]: why? Now you are returning promise instead of its value..

File: src/actions/QuestionsActions.js (hunk 1 above)
Comment (xgarrett) [inline src/actions/QuestionsActions.js]: yea, I changed it locally yesterday and forgotten to push :tada: Sorry :)

File: src/components/AskQuestion/index.js (hunk 2 above)
Comment (xgarrett) [inline src/components/AskQuestion/index.js]: Actually I don't know. :) Probably I was fighting against stateless functional component vs class.
But sure when I see this now with fresh look It should be Class with no excuses.

File: src/components/Question/icon_question.svg (hunk 3 above)
Comment (xgarrett) [inline src/components/Question/icon_question.svg]: svg is svg dude :P I just downloaded it from internet. Not sure if svg can have new line at end of file. Still it won't be changed anytime so I vote to leave it as it is.

File: src/components/Question/index.js (hunk 4 above)
Comment (xgarrett) [inline src/components/Question/index.js]: We need to talk about that. I'm not sure if it's always good as that.

File: src/utils/requester.js (hunk 5 above)
Comment (xgarrett) [inline src/utils/requester.js]: ouch so far, far ago... Probably very old change and shouldn't be here.

File: src/components/Question/icon_question.svg (hunk 3 above)
Comment (FredZgred) [inline src/components/Question/icon_question.svg]: haha yep. you're right

**Pull request merged**